front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -84,76 +80,12 @@ const TimesStops = ({ pathProperties, pathSteps = [], startTime }: TimesStopsPro
     setTimesStopsSteps(suggestedOPs);
   }, [t, pathProperties.allWaypoints, startTime]);
 
-  const columns: Column<PathWaypointColumn>[] = useMemo(
-    () => [
-      {
-        ...keyColumn<PathWaypointColumn, 'name'>('name', createTextColumn()),
-        title: t('name'),
-        disabled: true,
-      },
-      {
-        ...keyColumn<PathWaypointColumn, 'ch'>('ch', createTextColumn()),
-        title: 'Ch',
-        disabled: true,
-        grow: 0.1,
-      },
-      {
-        ...keyColumn<PathWaypointColumn, 'arrival'>('arrival', timeColumn),
-        title: t('arrivalTime'),
-
-        // We should not be able to edit the arrival time of the origin
-        disabled: ({ rowIndex }) => rowIndex === 0,
-        grow: 0.6,
-      },
-      {
-        ...keyColumn<PathWaypointColumn, 'stopFor'>(
-          'stopFor',
-          createTextColumn({
-            continuousUpdates: false,
-            alignRight: true,
-          })
-        ),
-        title: `${t('stopTime')}`,
-        grow: 0.6,
-      },
-      {
-        ...keyColumn<PathWaypointColumn, 'onStopSignal'>(
-          'onStopSignal',
-          checkboxColumn as Partial<Column<boolean | undefined>>
-        ),
-        title: t('receptionOnClosedSignal'),
-
-        // We should not be able to edit the reception on close signal if stopFor is not filled
-        // except for the destination
-        grow: 0.6,
-        disabled: ({ rowData, rowIndex }) =>
-          rowIndex !== pathProperties.allWaypoints?.length - 1 && !rowData.stopFor,
-      },
-      {
-        ...keyColumn<PathWaypointColumn, 'theoreticalMargin'>(
-          'theoreticalMargin',
-          createTextColumn({
-            continuousUpdates: false,
-            alignRight: true,
-            placeholder: t('theoreticalMarginPlaceholder'),
-            formatBlurredInput: (value) => {
-              if (!value || value === 'none') return '';
-              if (!marginRegExValidation.test(value)) {
-                return `${value}${t('theoreticalMarginPlaceholder')}`;
-              }
-              return value;
-            },
-          })
-        ),
-        cellClassName: ({ rowData }) => (rowData.isMarginValid ? '' : 'invalidCell'),
-        title: t('theoreticalMargin'),
-        disabled: ({ rowIndex }) => rowIndex === pathProperties.allWaypoints.length - 1,
-      },
-    ],
+  const columns = useMemo<Column<PathWaypointColumn>[]>(
+    () => inputColumns(t, pathProperties),
     [t, pathProperties.allWaypoints.length]
   );
 
-  const removeVia = (rowData: PathWaypointColumn) => {
+  const removeVia = (rowData: PathWaypointColumn) => () => {
```

Comment:
I find it less clear, a function returning a function. Do you think we could leave it like it was ?

Reply:
done